app/services/trainees/create_from_apply.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module Trainees
+  class CreateFromApply
+    include ServicePattern
+
+    def initialize(application:)
+      @application = application
+    end
+
+    def call
+      trainee.save!
+      trainee
+    end
+
+  private
+
+    attr_reader :application
+
+    def attributes
+      @attributes ||= parsed_application["attributes"]
+    end
+
+    def parsed_application
+      @parsed_application ||= JSON.parse(application.application)
+    end
+
+    def trainee
+      @trainee ||= Trainee.new(mapped_attributes)
+    end
+
+    def mapped_attributes
+      {
+        apply_application: application,
+        provider: application.provider,
+        trainee_id: raw_trainee["id"],
+        first_names: raw_trainee["first_name"],
+        last_name: raw_trainee["last_name"],
+        date_of_birth: raw_trainee["date_of_birth"],
+        gender: gender,
+        ethnic_group: ethnic_group,
+        ethnic_background: raw_trainee["ethnic_background"],
+        diversity_disclosure: diversity_disclosure,
+        email: raw_contact_details["email"],
+        # TODO: Use actual route type, if found.
+        training_route: 1,
+        subject: raw_course["course_code"],
```

Comment:
I don't think `course_code` is right here is it?

Reply:
As @gpeng  mentions, the trainee id is meant to be their own internal number - if storing an id from apply that should be kept separate.

Reply:
Changed to reference the course name in our course table, if present.